test/e2e/messaging.test.ts

```diff
@@ -0,0 +1,154 @@
+import { expect } from 'chai'
+import request from 'supertest'
+import { describe } from 'mocha'
+import { KeyType, ConnectionRecord } from '@aries-framework/core'
+import { getConfig } from './fixtures/config.js'
+
+import { pollGetConnections } from './utils/shared.js'
+
+describe('Messaging', async function () {
+  const config = getConfig()
+  const aliceClient = request(config.alice.portalUrl)
+  const bobClient = request(config.bob.portalUrl)
+  const aliceVeritableClient = request(config.alice.veritableUrl)
+  const bobVeritableCLient = request(config.bob.veritableUrl)
+  let bobTestDid: string | undefined = undefined
+  let aliceTestDid: string | undefined = undefined
+
+  before(async function () {
+    // import dids on Alice and Bob
+    await request(config.alice.veritableUrl)
+      .post('/dids/import')
+      .send({
+        did: config.alice.did,
+        privateKeys: [
+          {
+            keyType: KeyType.Ed25519,
+            privateKey: config.alice.privateKey,
+          },
+        ],
+        overwrite: true,
+      })
+      .set('Accept', 'application/json')
+      .expect(200)
+
+    await request(config.bob.veritableUrl)
+      .post('/dids/import')
+      .send({
+        did: config.bob.did,
+        privateKeys: [
+          {
+            keyType: KeyType.Ed25519,
+            privateKey: config.bob.privateKey,
+          },
+        ],
+        overwrite: true,
+      })
+      .set('Accept', 'application/json')
+      .expect(200)
+  })
+
+  after(async function () {
+    // clear all connections
+    const { body: aliceConns } = await aliceVeritableClient.get(`/connections`)
+    for (const { id } of aliceConns) {
+      await aliceVeritableClient.delete(`/connections/${id}`)
+    }
+    const { body: bobConns } = await bobVeritableCLient.get(`/connections`)
+    for (const { id } of bobConns) {
+      await bobVeritableCLient.delete(`/connections/${id}`)
+    }
+  })
+
+  describe('happy path', async function () {
+    it('Alice creates a connection with Bob & sends him a message.', async function () {
+      await aliceClient
+        .post('/connection')
+        .send({ did: config.bob.did })
+        .expect(204)
+
+      // poll until a completed connection appears
+      await pollGetConnections(
+        config.alice.veritableUrl,
+        (connections) =>
+          connections.length === 1 && connections[0].state === 'completed'
+      )
+
+      const { body: aliceConnections } = await request(
+        config.alice.veritableUrl
+      ).get('/connections')
+
+      const aliceConnectionRecord = aliceConnections[0] as ConnectionRecord
```

Comment:
This second call (L#77-84) is unnecessary if we assign `aliceConnectionRecord` inside the polling loop (L#74-75).